src/runtime/sg.c

```diff
@@ -38,6 +38,22 @@ static inline void sg_unlock(void)
 #define sg_unlock()
 #endif
 
+void sg_consume(sg_list sg, u64 length)
+{
+    sg_list_foreach(sg, sgb) {
+        if (length + sgb->offset >= sgb->size) {
+            sg_list_head_remove(sg);
+            length -= sgb->size - sgb->offset;
```

Comment:
Another nit, but this ```sgb->size - sgb->offset``` expression appears frequently...perhaps it could be replaced with something more descriptive like sg_buf_len() or sg_buf_remain().

Reply:
Replaced occurrences of `sgb->size - sgb->offset` with calls to an sg_buf_len() inline function.